### PR TITLE
[ADD] estate: create module estate

### PR DESCRIPTION
**Pull request by viso-odoo**
Create empty shell in the context of viso's onboarding as part of the R&D BE dev team

### CODE REVIEW

File: estate/models/estate_propery.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                record.state = 'sold'
+        return True
+
+    def action_set_cancel(self):
+        for record in self:
+            if record.state == 'sold':
+                raise UserError(_("Sold properties cannot be canceled"))
+            else:
+                record.state = 'canceled'
+        return True
```
Comment (vava-odoo) [inline estate/models/estate_propery.py:79]: This function could be simplified: avoid loop, prefer batch operations when possible
```suggestion
        if 'sold' in self.mapped('state'):
            raise UserError(_("Sold properties cannot be canceled."))
        self.state = 'canceled'
        return True
```

File: estate/security/ir.model.access.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+estate.access_estate_property,access_estate_property,estate.model_estate_property,base.group_user,1,1,1,1
+estate.access_estate_property_type,access_estate_property_type,estate.model_estate_property_type,base.group_user,1,1,1,1
+estate.access_estate_property_tag,access_estate_property_tag,estate.model_estate_property_tag,base.group_user,1,1,1,1
+estate.access_estate_property_offer,access_estate_property_offer,estate.model_estate_property_offer,base.group_user,1,1,1,1
+
```
Comment (vava-odoo) [inline estate/security/ir.model.access.csv:6]: One empty line is enough

File: estate/views/estate_menus.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    <menuitem id="real_estate_root" name="Real Estate">
+        <menuitem id="advertisements" name="Advertisements">
+            <menuitem id="real_estate_action" action="estate_property_action" />
+        </menuitem>
+        <menuitem id="settings" name="Settings">
+            <menuitem id="property_type_action" action="estate_property_type_action" />
+            <menuitem id="property_tag_action" action="estate_property_tag_action" />
+        </menuitem>
+    </menuitem>
+</odoo>
```
Comment (vava-odoo) [inline estate/views/estate_menus.xml:12]: but 0 is not enough :cry:

File: estate/views/estate_property_tag_views.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="estate_properties_tag_form" model="ir.ui.view">
+        <field name="name">estate.property.tag.form</field>
+        <field name="model">estate.property.tag</field>
+        <field name="arch" type="xml">
+            <form string="Tag">
+                <h1>
+                    <field name='name' />
+                </h1>
```
Comment (vava-odoo) [inline estate/views/estate_property_tag_views.xml:10]: That way, the label doesn't show up. Maybe better enclose it in a `<div oe_title>`

File: estate/views/estate_property_views.xml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            <search string="Property">
+                <field name="name" string="Property" />
+                <field name="postcode" />
+                <field name='expected_price' />
+                <field name='bedrooms' />
+                <field name='living_area' />
+                <field name='facades' />
+                <field name="property_type_id" />
+                <filter string="Available" name="available"
+                    domain="['|', ('state', '=', 'new'), ('state', '=', 'offer_received')]" />
```
Comment (vava-odoo) [inline estate/views/estate_property_views.xml:16]: Easier to read (and shorter)
```suggestion
                    domain="('state', 'in', ['new', 'offer_received'])]" />
```

File: estate/views/estate_property_views.xml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                            <field name='garage' />
+                            <field name='garden' />
+                            <field name='garden_area' />
+                            <field name='garden_orientation' />
+                            <field name="total_area" />
+                        </group>
+                    </page>
+                    <page string="Offers">
+                        <field name="offer_ids" context="{'tree_view':'estate_property_offer_tree'}"
+                            mode="tree" />
```
Comment (vava-odoo) [inline estate/views/estate_property_views.xml:68]: Do you need all this?
```suggestion
                        <field name="offer_ids"/>
```

File: estate/models/estate_property_offer.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    )
+    partner_id = fields.Many2one('res.partner', string="Partner", required=True)
+    property_id = fields.Many2one('estate.property', required=True)
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline", string="Deadline")
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for offer in self:
+            tmp_date = fields.Date.today()
```
[the remark this answers is not in the document]
Comment (viso-odoo) [reply estate/models/estate_property_offer.py:28]: but if I do this, tmp_date is not reset for the next record leading to potentially weird behaviour

File: estate/models/estate_property_offer.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    status = fields.Selection(
+        selection=[
+            ("accepted", "Accepted"),
+            ("refused", "Refused")
+        ],
+        copy=False
+    )
+    partner_id = fields.Many2one('res.partner', string="Partner", required=True)
+    property_id = fields.Many2one('estate.property', required=True)
+    validity = fields.Integer(default=7)
```
[the remark this answers is not in the document]
Comment (viso-odoo) [reply estate/models/estate_property_offer.py:18]: Updated

File: estate/models/estate_property_offer.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (viso-odoo) [reply estate/models/estate_property_offer.py:28]: made another kind of change based on a later comment, seems to work fine \side eye\

File: estate/models/estate_property_offer.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline", string="Deadline")
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for offer in self:
+            tmp_date = fields.Date.today()
+            if offer.create_date:
+                tmp_date = offer.create_date
```
[the remark this answers is not in the document]
Comment (viso-odoo) [reply estate/models/estate_property_offer.py:30]: updated

File: estate/models/estate_property_offer.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+            offer.date_deadline = fields.Date.add(tmp_date, days=offer.validity)
+
+    def _inverse_date_deadline(self):
+        for offer in self:
+            start_date = fields.Date.from_string(offer.create_date)
```
[the remark this answers is not in the document]
Comment (viso-odoo) [reply estate/models/estate_property_offer.py:36]: added `or fields.Date.today()`

File: estate/views/estate_property_tag_views.xml (hunk 4 above)
Comment (viso-odoo) [reply estate/views/estate_property_tag_views.xml:10]: put a placeholder instead, but it looks cool

File: estate/views/estate_property_views.xml (hunk 6 above)
Comment (viso-odoo) [reply estate/views/estate_property_views.xml:68]: apparently not :laughing:

**Pull request closed without merging**